### PR TITLE
Update calloop to v0.9.0

### PR DESCRIPTION
**Pull request by cmeissl**
This PR updates calloop to v0.9.0 and fixes the remaining warnings and clippy errors

### CODE REVIEW

File: src/event_loop.rs [hunk 1]
```diff
@@ -111,19 +112,23 @@ impl EventSource for WaylandSource {
             // quickly. Either it is slowed down or we are a spammer.
             // Should not really happen, if it does we do nothing and will flush again later
         }
-        Ok(())
+        Ok(PostAction::Continue)
```
Comment (vberger) [inline src/event_loop.rs:115]: Due to the new semantics of calloop, for full correctness the whole body of this function needs to be wrapped En the callback of `self.fd.process_events()`.

Comment (cmeissl) [reply src/event_loop.rs]: Ok, moved into `self.fd.process_events`.

Comment (cmeissl) [reply src/event_loop.rs]: Are you ok with the changes or is there anything else that needs to be changed?

**Comment by vberger** [comment]
Thanks! However it looks like some imports are breaking the CI. I think you need to add a few `#[cfg()]` attributes so that they are only imported when the feature is enabled.

**Comment by l4l** [comment]
`calloop` updates nix package to 0.21 ([link](https://github.com/Smithay/calloop/blob/master/Cargo.toml#L19)), would be handy to update there as well.

**Comment by vberger** [comment]
Actually there is also `0.22` pending as well.

**Comment by cmeissl** [comment]
> Thanks! However it looks like some imports are breaking the CI. I think you need to add a few `#[cfg()]` attributes so that they are only imported when the feature is enabled.

Should be fixed now.

**Pull request merged**